### ISSUE TITLE
Veusz.embed functions sometimes hangs (not returns)

### ISSUE THREAD

**Issue by And0k** [veusz/veusz#223]
I have heavy data Veusz files. Frequently can not export jpg/png using Python 3.5 64bit, Veusz 2.1.1. Same noticed with function IsClosed(). But export works if load them normally (manually) - take till 15 seconds per page

**Comment by jeremysanders** [comment]
I'll try to look at this later, but if you can write a Python script demonstrating this, it would be great.

**Comment by jeremysanders** [comment]
I can't reproduce this with a 1000000 point plot.

**Comment by And0k** [comment]
[171017.zip](https://github.com/veusz/veusz/files/1533711/171017.zip)
- this is my file with data. Usually hangs when export page #5.

**Comment by jeremysanders** [comment]
Nice plot! If I load this into an embedded instance doing

    e = veusz.Embedded('test')
    e.Load('171017#01.vsz')
    e.Export('test.png')

Then that Export returns within a couple of seconds for me (also with jpg). IsClosed() is ok. I've tried python2 and 3 on Ubuntu artful. What operating system are you using?

It looks like the CSV reader speed could be improved, but this would probably require a rewrite into C++.

**Comment by And0k** [comment]
Yes, 1st page usually exports Ok. Moreover I've compiled Veusz on Fedora 26, tried run my program there and all 20 Veusz files have been exported without problem, so this is Windows specific problem

**Comment by jeremysanders** [comment]
I tried it on my windows setup, and it seems to export all the pages ok. Are you usuing source code or the binary?

It's possible that a firewall or virus scanner might break the program. It uses a local network socket to do the communication between the Veusz server and Python module. Maybe procmon could be used to identify what the two processes are doing when it hangs: https://docs.microsoft.com/en-us/sysinternals/downloads/procmon

**Comment by And0k** [comment]
Now tries export images from 500 vsz files. After each 50 Embedded.Load() calls this command hangs. Solved the issue by timeout and reimport _embed_ from _veusz_
Debugging hangs in Veusz 3.2.1, Windows 10 shows that socket tries receive very big message.
